enumerator.c

```diff
@@ -3802,6 +3802,13 @@ rb_arithmetic_sequence_beg_len_step(VALUE obj, long *begp, long *lenp, long *ste
     *stepp = step;
 
     if (step < 0) {
+        if (aseq.exclude_end && !NIL_P(aseq.end)) {
+            /* Handle exclusion before range reversal */
+            aseq.end = LONG2NUM(NUM2LONG(aseq.end) + 1);
+
+            /* Don't exclude the previous beginning */
+            aseq.exclude_end = 0;
+        }
```

Comment:
We need to write tests of `rb_arithmetic_sequence_beg_len_step` at least to check the behavior of this case because this is a public C-API.
Could you please write `ext/-test-/arith_seq/beg_len_step` and `test/-ext-/arith_seq/test_beg_len_step.rb`?

Reply:
I added tests for rb_arithmetic_sequence_beg_len_step as requested.